Jetcaster/tv-app/src/main/java/com/example/jetcaster/tv/ui/search/SearchScreenViewModel.kt

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.jetcaster.tv.ui.search
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.jetcaster.core.data.database.model.Category
+import com.example.jetcaster.core.data.di.Graph
+import com.example.jetcaster.core.data.repository.CategoryStore
+import com.example.jetcaster.core.data.repository.PodcastStore
+import com.example.jetcaster.core.data.repository.PodcastsRepository
+import com.example.jetcaster.tv.model.CategoryList
+import com.example.jetcaster.tv.model.CategorySelection
+import com.example.jetcaster.tv.model.CategorySelectionList
+import com.example.jetcaster.tv.model.PodcastList
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+
+class SearchScreenViewModel(
+    private val podcastsRepository: PodcastsRepository = Graph.podcastRepository,
+    private val podcastStore: PodcastStore = Graph.podcastStore,
+    categoryStore: CategoryStore = Graph.categoryStore,
+) : ViewModel() {
+
+    private val keywordFlow = MutableStateFlow("")
+    private val selectedCategoryListFlow = MutableStateFlow<List<Category>>(emptyList())
+
+    private val categoryListFlow = categoryStore.categoriesSortedByPodcastCount().map {
+        CategoryList(it)
+    }
+
+    private val searchConditionFlow =
+        combine(keywordFlow, selectedCategoryListFlow) { keyword, selectedCategories ->
+            SearchCondition(keyword, selectedCategories)
+        }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private val searchResultFlow = searchConditionFlow.flatMapLatest {
+        podcastStore.searchPodcastByTitleAndCategories(it.keyword, it.selectedCategories)
```

Comment:
I believe this will still trigger a search when the keyword is the empty string. Do we want that (i.e. still show results if there is no input)? Or should the empty case be handled another way?

Reply:
It is an intended behavior to trigger the search with empty string. I expect that all podcasts belonging to the selected category show up when the empty string is input.